### ISSUE TITLE
Failed to scheduling EKS pods: running PreBind plugin "VolumeBinding": binding volumes: timed out waiting for the condition

### ISSUE THREAD

**Issue by thbkrkr** [elastic/cloud-on-k8s#6515]
[e2e-tests-eks-arm/632](https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-eks-arm/632/testReport/github/com_elastic_cloud-on-k8s_v2_test_e2e_agent/TestSystemIntegrationRecipe_All_expected_Pods_should_eventually_be_ready/)

EKS pods cannot be scheduled because they are waiting for their volumes to be bound.

In #6469, we updated EKS to install k8s `1.24` but we now also need to install the Amazon EBS CSI driver:

> If you plan to deploy workloads that use Amazon EBS volumes in a new 1.23 cluster, install the Amazon EBS CSI driver in your cluster before deploying the workloads your cluster. 
(https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-1.23).

```json
{
    "apiVersion": "v1",
    "kind": "Event",
    "count": 3,
    "firstTimestamp": "2023-03-13T01:25:07Z",
    "lastTimestamp": "2023-03-13T01:45:11Z", 
   "involvedObject": {
        "apiVersion": "v1",
        "kind": "Pod",
        "name": "elasticsearch-wdrv-es-default-0",
        "namespace": "e2e-sdroq-mercury",
    },
    "message": "running PreBind plugin \"VolumeBinding\": binding volumes: timed out waiting for the condition",
    "reason": "FailedScheduling",
    "type": "Warning",
    "source": {
        "component": "default-scheduler"
    }
}
```

**Comment by pebrc** [comment]
Are we not using local volumes for all tests? Did the disk setup not work?

**Comment by thbkrkr** [comment]
I didn't question that, I'm very unsure but it seemed to me that the majority of the tests use local volumes but that we purposely kept some tests with EBS?

According to the log, the disk setup looks fine:
```
[2023-03-13T01:03:39.937Z] Added new context arn:aws:eks:eu-west-1:444732909647:cluster/eck-jks-e2e-eks-arm-632 to /root/.kube/config
[2023-03-13T01:03:43.046Z] storageclass.storage.k8s.io/e2e-default created
[2023-03-13T01:03:43.447Z] configmap/local-provisioner-config created
[2023-03-13T01:03:43.950Z] configmap/setup-disks created
[2023-03-13T01:03:44.352Z] daemonset.apps/local-volume-provisioner created
[2023-03-13T01:03:44.753Z] serviceaccount/local-storage-admin created
[2023-03-13T01:03:45.255Z] clusterrolebinding.rbac.authorization.k8s.io/local-storage-provisioner-pv-binding created
[2023-03-13T01:03:45.657Z] clusterrole.rbac.authorization.k8s.io/local-storage-provisioner-node-clusterrole created
[2023-03-13T01:03:46.158Z] clusterrolebinding.rbac.authorization.k8s.io/local-storage-provisioner-node-binding created
[2023-03-13T01:03:47.362Z] NAME            PROVISIONER                    RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
[2023-03-13T01:03:47.362Z] e2e-default     kubernetes.io/no-provisioner   Delete          WaitForFirstConsumer   false                  5s
[2023-03-13T01:03:47.362Z] gp2 (default)   kubernetes.io/aws-ebs          Delete          WaitForFirstConsumer   false                  7m51s
```

The failing test is `TestSystemIntegrationRecipe` which is based on the `system-integration.yaml` recipe. I don't see anything to enforce the use of the `e2e-default` storageClass. It would seem that all tests that use a recipe create ES clusters with PersistentVolumeClaims set with the default storage class configured by the cloud provider and not ours `e2e-default`.

**Comment by thbkrkr** [comment]
This is clearly the problem here, the PVCs are created with the `gp2` storage class defined with the `kubernetes.io/aws-ebs` provisioner.

```json
{
    "apiVersion": "v1",
    "kind": "PersistentVolumeClaim",
    "metadata": {
        "annotations": {
            "volume.beta.kubernetes.io/storage-provisioner": "ebs.csi.aws.com",
            "volume.kubernetes.io/selected-node": "ip-192-168-60-241.eu-west-1.compute.internal",
            "volume.kubernetes.io/storage-provisioner": "ebs.csi.aws.com"
        },
        "name": "elasticsearch-data-elasticsearch-wdrv-es-default-0",
        ... 
    },
    "spec": {
        "accessModes": [
            "ReadWriteOnce"
        ],
        "resources": {
            "requests": {
                "storage": "1Gi"
            }
        },
        "storageClassName": "gp2",
        "volumeMode": "Filesystem"
    },
    "status": {
        "phase": "Pending"
    }
}
```

**Comment by thbkrkr** [comment]
It's not clear to me if this is a real intent or a bug that we have tests that use the cloud providers default storage class and therefore mostly require storage in network?

I tried to do a little archeology to find our intention.

Oct 11, 2019 We added AKS to the tests matrix and had to create a non-default storage class for e2e tests (https://github.com/elastic/cloud-on-k8s/pull/1974).

Oct 22, 2019 We wrote the e2e-tests yaml reader to reuse external manifests to test Kibana standalone (https://github.com/elastic/cloud-on-k8s/pull/2021).

Dec 9, 2019, we started by improving the execution time of the e2e tests on GKE by using the local SSDs instead of attached GCP persistent disks (https://github.com/elastic/cloud-on-k8s/pull/2227).

We then agreed to continue running the examples in `config/samples` with the default storage class to validate that ECK is still compatible with GKE's default storage class (https://github.com/elastic/cloud-on-k8s/pull/2227#issuecomment-563133586). You had then precisely made a very good comment which beats us today:
> I would like to have a more explicit way of testing network attached storage than just the samples. But that can come in an another PR.

And we added more recipes and tests that use those recipes omitting that they used network attached storage.

**Comment by thbkrkr** [comment]
> I guess there is some value in having a few network attached tests but I don't feel super strongly about it tbh.

No strong opinion either. However, I think there is quick-win to (at least temporarily) solve this with a oneliner that forces the use of our storage class and therefore stops testing EBS.

**Comment by ricardorqr** [comment]
Hello @thbkrkr,

Have you solved this problem? I'm facing the same problem.

`running PreBind plugin "VolumeBinding": binding volumes: timed out waiting for the condition `

**Comment by thbkrkr** [comment]
@ricardorqr Are you using EBS, k8s >= 1.23 without installing the Amazon EBS CSI driver?

**Comment by ricardorqr** [comment]
> @ricardorqr Are you using EBS, k8s >= 1.23 without installing the Amazon EBS CSI driver?

Yes, that is correct. My Kubernetes cluster is 1.24 and I'm not using EBS. While searching I found this link

https://repost.aws/knowledge-center/eks-persistent-storage

Where teaches how to install EBS CSI driver. However, I don't know which one I should install. EBS or EFS?
